base/number.jl

```diff
@@ -133,11 +133,41 @@ zero{T<:Number}(::Type{T}) = convert(T,0)
 """
     one(x)
 
-Get the multiplicative identity element for the type of `x` (`x` can also specify the type
-itself). For matrices, returns an identity matrix of the appropriate size and type.
+Return a multiplicative identity for `x` (which can be either a value
+or a type), of the same shape and numeric precision as `x`.  (For matrices,
```

Comment:
I'm pretty sure that most people disliked the "shape" part, and that most think thought `one` on `Matrix{Float64}` should return a number (1.0, not `eye(n,n)`).

This PR doesn't have to do the deprecation, but I don't think we should put "shape" into the docstring.